compositor_common/src/frame.rs

```diff
@@ -0,0 +1,14 @@
+use crate::scene::Resolution;
+
+#[allow(dead_code)]
+pub struct Frame {
+    pub data: YuvData,
+    pub resolution: Resolution,
+    pub pts: i64,
```

Comment:
Negative presentation timestamps are really weird here. I don't know exactly why FFmpeg starts producing negative frames, but I think we should fix it somehow. I found a similar issue with the patch solution described [here](https://stackoverflow.com/questions/60264279/ffmpeg-returns-negative-pts-and-dts-for-first-non-key-frame). 

Overall, we probably want to receive streams starting with PTS 0 and give users an option to add some sort of `pts_offset` (to synchronize newly added videos with ones added earlier). I think we should use `u64` as `pts` to make invalid states unrepresentable.

Reply:
I agree with the point that we should avoid negative values, but I think pts should still be represented by a signed integer.
- ffmpeg frame/packet is using a signed value for pts.
- If I understand correctly negative pts is valid (even though it's not desirable)
- It does not complicate offset logic

> Overall, we probably want to receive streams starting with PTS 0 

Should this be handled here or in queue? On the one hand probably it would be easier to do it here, but if want to use pipeline create as standalone composer in the future, it might make sense to move offset synchronizations to the queue, so we don't reimplement that.

Reply:
In the queue, the code handling new streams will be the same. I don't understand what negative PTS should even mean in our case, so I would use `u64` here. Even though the queue will handle negative timestamps as well, I think that the output of the `RtpReceiver` should always return increasing timestamps starting from 0.